internal/newusers/manager.go

```diff
@@ -278,6 +295,46 @@ func (m *Manager) startUserCleanupRoutine(cacheDir string, opts *options) {
 	<-cleanupRoutineStarted
 }
 
+// isDbCorrupted checks if the database is marked as corrupted.
+func (m *Manager) isDbCorrupted() bool {
+	_, err := os.Stat(m.dirtyFlagPath)
+	return err == nil
+}
+
+// markCorruptedDb writes a dirty flag in the cache directory to mark the database as corrupted.
+func (m *Manager) markCorruptedDb() error {
```

Comment:
Also, similar `markCorrupted`. Btw, something it’s `dbCorrupted`, somettimes `CorruptedDb`, that would ensure a single naming convention :)